repo_processing/tree_sitter/tree_sitter_parser.py

```diff
@@ -0,0 +1,175 @@
+
+from repo_processing.dulwich.enry_parser import get_content
+from repo_processing.dulwich.extract import clone_or_instantiate
+
+from tree_sitter import Language, Parser, Tree
+from tree_sitter import Node
+
+# query of imports, used methods, classes
+java_imports_used_methods_query_string = """
+(import_declaration ((scoped_identifier (identifier)) (asterisk)) @name)
+(package_declaration  (identifier) @name)  
+(method_invocation name: (identifier) @function.call) 
+(local_variable_declaration declarator: (variable_declarator name: (identifier) @name)) 
+"""
+
+# query of names of created packages, classes, interfaces, methods, arguments, fields
+java_names_query = """
+(package_declaration (scoped_identifier (identifier)) @name)
+(class_declaration name: (identifier) @name)
+(interface_declaration name: (identifier) @name)
+(method_declaration name: (identifier) @name)  
+(method_declaration (formal_parameters(formal_parameter (identifier) @name))) 
+(field_declaration declarator: (variable_declarator name: (identifier) @name))
+"""
+
+# query of imports, called functions
+python_imports_used_methods_query_string = """
+(import_from_statement (dotted_name (identifier)) @dotted_name)
+(import_statement (dotted_name (identifier)) @dotted_name)
+(aliased_import (dotted_name (identifier)) @dotted_name)
+(call function: (identifier) @function.call)
+"""
+
+# query of names of classes, functions, fields
+python_names_query_string = """
+(class_definition name: (identifier) @name)
+(function_definition name: (identifier) @function.def)
+(expression_statement (assignment left: (identifier) @name)) 
+"""
+
+# query of imports, called functions
+js_imports_used_methods_query_string = """
+(import_statement (import_clause (identifier) @name) (string (string_fragment)) @name)
+(import_statement (import_clause (namespace_import (identifier) @name)) (string (string_fragment) @name))
+(import_statement (import_clause (named_imports (import_specifier (identifier)) @name))(string (string_fragment) @name))
+(import_statement (import_clause (named_imports (import_specifier (identifier)) (import_specifier (identifier) 
+    (identifier) @name))) (string (string_fragment)))
+(call_expression function: (identifier) @function arguments: (arguments (identifier))) @function.call
+"""
+
+# query of names of classes, functions, fields
+js_names_query_string = """(assignment_expression (identifier) @name) (variable_declaration (variable_declarator (
+identifier) @name)) (lexical_declaration (variable_declarator (identifier) @name)) (expression_statement (
+call_expression (member_expression (call_expression (import) (arguments (string (string_fragment) @name))) (
+property_identifier)) (arguments (arrow_function (formal_parameters (identifier) @name) (statement_block))))) (
+expression_statement (member_expression) @name) (class_declaration name: (identifier) @name) (function_declaration 
+    name: (identifier) @name) """
+
+
+def setup_tree_sitter_parser() -> None:
```

Comment:
please add logic that will check if parsers are ready - if not call this function in this file - so any time you import from this file - parsers will be checked that they are available